### DISCUSSION TITLE
Question: Reserved words and quoting

### DISCUSSION THREAD

**Discussion by boesing** [doctrine/doctrine2#11047] [Support Questions]
|    Q        |   A
|------------ | -----
| Version     | any

### Support Question

I've stumbled over https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/basic-mapping.html#quoting-reserved-words which says that one has to quote columns (and I guess also table names?) since doctrine does not quote these on its own.

What I wonder is, what is the purpose of the reserved words per PHP version then if its not to quote stuff which is reserved.
Is that documentation still up2date or did something change meanwhile?

Does this also apply to table names for MySQL 8?
```php
#[Table(name:"`lead`")]
```

https://github.com/doctrine/dbal/blob/45941c67dd0505dab2fb970786d93055b7cbd2b6/src/Platforms/Keywords/MySQL80Keywords.php#L55

**Comment by derrabus** [comment]
The documentation is up to date, as far as I know. You are right though that the DBAL provides us with a list of keywords that need quoting. The ORM could theoretically leverage that. Do you want to give it a try?

**Reply by boesing** [reply]
Yah, I will see if and when I can implement that.
